app/models/assignment_participant.rb

```diff
@@ -107,33 +107,30 @@ def team
     AssignmentTeam.team(self)
   end
 
-  # provide import functionality for Assignment Participants
-  # if user does not exist, it will be created and added to this assignment
-
-  def self.import(row_hash, _row_header = nil, session, id)
-    raise ArgumentError, 'No user id has been specified.' if row_hash.empty?
-
-    user = User.find_by(name: row_hash[:username])
-
-    # if user with provided name in csv file is not present then new user will be created.
-    if user.nil?
-      raise ArgumentError, "The record containing #{row_hash[:username]} does not have enough items." if row_hash.length < 4
-
-      # define_attributes method will return an element that stores values from the row_hash.
-      attributes = ImportFileHelper.define_attributes(row_hash)
-
-      # create_new_user method will create new user with values present in attribute.
-      user = ImportFileHelper.create_new_user(attributes, session)
-
+  # Note: If user does not exist, it will be created and added to this assignment
+  def self.import(row_hash, session, id)
+    raise ArgumentError, 'Record does not contain required items.' if row_hash.length < required_import_fields.length
+    user = User.find_by(name: row_hash[:name])
+    user = User.import(row_hash, session, nil) if user.nil?
+    raise ImportError, "The assignment with id #{id} was not found." if Assignment.find(id).nil?
+    unless AssignmentParticipant.exists?(user_id: user.id, parent_id: id)
+      new_part = AssignmentParticipant.new(user_id: user.id, parent_id: id)
+      new_part.set_handle
     end
-    raise ImportError, "The assignment with id \"#{id}\" was not found." if Assignment.find(id).nil?
+  end
 
-    # if user is already added to the assignment then return.
-    return if AssignmentParticipant.exists?(user_id: user.id, parent_id: id)
+  def self.required_import_fields
```

Comment:
Another thing I don't understand: why is this code duplicated in assignment_participant and course_participant, instead of being located in participant?

Reply:
I have abstracted the import field methods into the participant.rb file.